src/main.js

```diff
@@ -1,23 +1,25 @@
+// @flow
+
 import { getDate, getDayOfYear, getDaysInMonth, getDaysInYear, isMonday, isSameDay, isSunday, subDays, subYears } from './helpers';
 import { applyOffset, formatDate, getDiffFn, getEndOfFn, getStartOfFn, getSubtractionFn } from './utils';
+import type { CompareMode, DateLike, Period, PeriodParams, Range, RangeAndLabel, Unit } from './types.js.flow';
 
-// :: (Date | Int | String) -> (Date | Int | String)  -> Object
-const formatRange = (start, end) => ({
+const formatRange = (start: DateLike, end: DateLike): Range => ({
   start: formatDate(start),
   end: formatDate(end),
 });
 
-// :: Int -> String -> Option(Date | Int | String) = new Date() -> Object
-const getLastRelativePeriodRange = (num, unit, base = new Date()) => {
+const getLastRelativePeriodRange = (num: number, unit: Unit, base: Date = new Date()): Range => {
   const startOf = getStartOfFn(unit);
   const endOf = getEndOfFn(unit);
   const sub = getSubtractionFn(unit);
 
   return formatRange(startOf(sub(base, num)), endOf(sub(base, 1)));
 };
 
-// :: String -> Option(Date | Int | String) = new Date() -> Object
-const getThisRelativePeriodRange = (unit, base = new Date(), num = 1) => {
+const getThisRelativePeriodRange = (
+  unit: Unit, base: Date = new Date(), num: number = 1,
```

Comment:
Add `?` to `base` and `num` as they are optional?

And should `base` be `DateLike` instead?